lib/LaTeXML/Core/Mouth.pm

```diff
@@ -159,24 +159,51 @@ sub stringify {
 
 #**********************************************************************
 sub getLocator {
-  my ($self, $length) = @_;
-  my ($l, $c) = ($$self{lineno}, $$self{colno});
-  if ($length && ($length < 0)) {
-    return "at $$self{shortsource}; line $l col $c"; }
-  elsif ($length && (defined $l || defined $c)) {
-    my $msg   = "at $$self{source}; line $l col $c";
-    my $chars = $$self{chars};
-    if (my $n = $$self{nchars}) {
-      $c = $n - 1 if $c >= $n;
-      my $c0 = ($c > 50      ? $c - 40 : 0);
-      my $cm = ($c < 1       ? 0       : $c - 1);
-      my $cn = ($n - $c > 50 ? $c + 40 : $n - 1);
-      my $p1 = ($c0 <= $cm ? join('', @$chars[$c0 .. $cm]) : ''); chomp($p1);
-      my $p2 = ($c <= $cn  ? join('', @$chars[$c .. $cn])  : ''); chomp($p2);
-      $msg .= "\n  " . $p1 . "\n  " . (' ' x ($c - $c0)) . '^' . ' ' . $p2; }
-    return $msg; }
-  else {
-    return "at $$self{source}; line $l col $c"; } }
+  # Make Xpointer style locator optional
+  $STATE->assignValue('LOCATOR_STYLE' => 'narrative') unless $STATE->lookupValue('LOCATOR_STYLE');
```

Comment:
This check should happen once in ```latexml.sty.ltxml``` rather than every time when getLocator() is called. The locator function is called extremely often, so we should make sure it does as little as possible.

Especially with the way you've handled the conditional, you can assume an unset value leads to the default.

So something like:
```perl
 my $locator_style = $STATE->lookupValue('LOCATOR_STYLE')
 if ($locator_style && ($locator_style eq 'xpointer')) {
   return get_locator_xpointer(@_);
 } else {
   return get_locator_narrative(@_); } }
```

suffices here.